app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -111,6 +119,40 @@ class MessageActivity : AppCompatActivity() {
                 flash.toggle()
         }
 
+        save.setOnClickListener {
+            try {
+                val inputManager: InputMethodManager = this.getSystemService(Context.INPUT_METHOD_SERVICE)
+                        as InputMethodManager
+                inputManager.hideSoftInputFromWindow(content.windowToken, InputMethodManager.SHOW_FORCED)
+
+                val sdf = SimpleDateFormat("dd-MMM-yyyy-hh_mm_aaa")
+                val fileName = "badge-magic-" + sdf.format(Calendar.getInstance().time).toString()
+                val os = FileOutputStream(fileName, true)
+
+                if (content.text.isNotBlank())
+                    moshiConverter.toStream(("Content-Text - " + content.text.toString()), os)
+                else if (drawableRecyclerAdapter.selectedPosition >= 0)
+                    moshiConverter.toStream(("Content-Drawable - " + drawableRecyclerAdapter.selectedPosition), os)
+
+                moshiConverter.toStream(speed.selectedItem.toString(), os)
+                moshiConverter.toStream(mode.selectedItem.toString(), os)
```

Comment:
Please create a POJO and serialize it instead of adding test line by line

Reply:
Ok I'll do that